File: backend/src/main/java/com/zola/services/notification/NotificationService.java
package com.zola.services.notification;

import com.zola.dto.response.notification.NotificationResponse;
import com.zola.entity.User;
import com.zola.enums.NotificationType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    Page<NotificationResponse> getUserNotifications(Pageable pageable);
    void markAsRead(Long notificationId);
    void markAllAsRead();
    long getUnreadCount();
    void createNotification(User user, String title, String message, NotificationType type);
    void notifyAdmins(String title, String message, NotificationType type);
}

File: frontend/src/services/notification.service.ts
import api from './api';

export enum NotificationType {
    ORDER = 'ORDER',
    PROMOTION = 'PROMOTION',
    SYSTEM = 'SYSTEM',
}

export interface NotificationResponse {
    id: number;
    title: string;
    message: string;
    type: NotificationType;
    read: boolean;
    createdAt: string;
}

export interface PageResponse<T> {
    content: T[];
    totalPages: number;
    totalElements: number;
    size: number;
    number: number;
}

const notificationService = {
    getUserNotifications: async (page = 0, size = 10): Promise<PageResponse<NotificationResponse>> => {
        const response = await api.get(`/notifications?page=${page}&size=${size}`);
        return response.data.result;
    },

    markAsRead: async (id: number): Promise<void> => {
        await api.patch(`/notifications/${id}/read`);
    },

    markAllAsRead: async (): Promise<void> => {
        await api.patch('/notifications/read-all');
    },

    getUnreadCount: async (): Promise<number> => {
        const response = await api.get('/notifications/unread-count');
        return response.data.result;
    },
};

export default notificationService;
